sdk/src/Services/DynamoDBv2/Custom/DataModel/TransactWrite.cs

```diff
@@ -143,6 +143,39 @@ public void AddSaveItem(T item)
             objectItems.Add(objectItem);
         }
 
+        /// <summary>
+        /// Add a single item to be saved in the current transaction operation.
+        /// Item is identified by its hash primary key and will be updated using the update expression provided.
+        /// </summary>
+        /// <param name="hashKey">Hash key of the item to delete.</param>
+        /// <param name="updateExpression">Update expression to use.</param>
+        /// <param name="conditionExpression">Condition to check before the operation.</param>
+        public void AddSaveKey(object hashKey, Expression updateExpression, Expression conditionExpression = null)
```

Comment:
Thinking about naming here, `Key` feels implied since all items will have one. I wonder if something like `AddSaveExpression` or `AddSaveViaExpression` is more meaningful relative to the existing `AddSaveItem`.

(not asking for a change yet, curious what you or other reviewers think)

Reply:
Hmm... I think we have two alternate paths to take here instead of `AddSaveKey`. The first is to include the term "expression" in the name, like `AddSave(Item)(Via|By|With)(Update)Expression`. The second is to call this a new operation entirely, e.g. `AddUpdateItem`. I'm happy with any of these, whatever you decide.

Reply:
Looking at it again, I think I like just overloading `AddSaveItem`. That looks like what Java does here https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/enhanced/dynamodb/model/WriteBatch.Builder.html

But not requesting changes yet, maybe a second opinion from @normj or @boblodgett since the reviewed this API originally?

Reply:
Discussed with the team internally today - could you adjust it to keep overloading `AddSaveItem`?

We feel that functionally we're just adding another action to the transaction, so lets keep the same operation name regardless of the input variations.

Reply:
Done